File: e2e-tests/spring-rest-openapi-v3/src/main/kotlin/com/foo/rest/examples/spring/openapi/v3/statistics/StatisticsRest.java
package com.foo.rest.examples.spring.openapi.v3.statistics;

import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping(path = "/api/statistics")
public class StatisticsRest {
    @RequestMapping(
            value = "/getExpectations/{b}",
            method = RequestMethod.GET
    )
    public String expectTest(
            @PathVariable("b") boolean succeeded
    ) throws IllegalArgumentException{
        if (succeeded) return "Success is True!";
        else throw new IllegalArgumentException("Input Was False");
    }

    @RequestMapping(
            value = "/basicResponsesString/{b}",
            method = RequestMethod.GET
    )
    public String basicResponsesString(
            @PathVariable("b") boolean succeeded
    ) {
        if (succeeded) return "Success is True!";
        else return "What do you mean no string?";
    }

    @RequestMapping(
            value = "/basicNumbersString/{b}",
            method = RequestMethod.GET
    )
    public int basicResponsesNumber(
            @PathVariable("b") boolean succeeded
    ) {
        if (succeeded) return 42;
        else return -1;
    }
}

File: core-tests/e2e-tests/spring-rest-openapi-v3/src/main/kotlin/com/foo/rest/examples/spring/openapi/v3/statistics/StatisticsRest.kt
package com.foo.rest.examples.spring.openapi.v3.statistics

import org.springframework.http.ResponseEntity
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController


@RestController
@RequestMapping(path = ["/api/statistics"])
private class StatisticsRest {

    @GetMapping(path = ["/getExpectations/{b}"])
    open fun expectTest(@PathVariable("b") succeeded: Boolean ) : ResponseEntity<String> {
        if (succeeded){
            return ResponseEntity.ok("Success!")
        }
        else{
            return ResponseEntity.status(500).build()
        }

    }

    @GetMapping(path = ["/basicResponsesString/{b}"])
    open fun basicResponseString(@PathVariable("b") succeeded: Boolean) : ResponseEntity<String> {
        if (succeeded){
            return ResponseEntity.ok("Success!")
        }
        else{
            return ResponseEntity.status(400).build()
        }
    }

    @GetMapping(path = ["/basicNumbersString/{b}"])
    open fun basicResponsesNumber(@PathVariable("b") succeeded: Boolean) : ResponseEntity<String> {
        if (succeeded){
            return ResponseEntity.ok("42")
        }
        else{
            return ResponseEntity.ok("-1")
        }
    }
}
